manifests/server/config.pp

```diff
@@ -161,18 +161,12 @@
   }
 
   if $handle_creds {
-    if $auth and $store_creds {
-      file { $rcfile:
-        ensure  => file,
-        content => template('mongodb/mongorc.js.erb'),
-        owner   => 'root',
-        group   => 'root',
-        mode    => '0600',
-      }
-    } else {
-      file { $rcfile:
```

Comment:
I think purging the file if it's not required anymore makes sense? shouldn't we keep it?

Reply:
The file contains functions that are used in the code regardless if auth is enabled or not.
ie: https://github.com/voxpupuli/puppet-mongodb/blob/master/templates/mongorc.js.erb#L1 contains the 'rsReconfigMember' function that is called by 'rs_reconfig_member' in https://github.com/voxpupuli/puppet-mongodb/blob/master/lib/puppet/provider/mongodb_replset/mongo.rb#L98 . If the mongorc.js file is not present (regardless if auth is enabled or not), it will crash if a rs member is reconfigured when auth is not used.